static/app/views/issueList/utils/useFetchIssueTags.tsx

```diff
@@ -0,0 +1,303 @@
+  start?: string;
+  statsPeriod?: string;
+  useCache?: boolean;
+};
+
+/**
+ * Fetches the tags from both the Errors and IssuePlatform dataset
+ * and combines them with builtin and predfined tags.
+ */
+export const useFetchIssueTags = ({
+  org,
+  projectIds,
+  keepPreviousData = false,
+  useCache = true,
+  enabled = true,
+  ...statsPeriodParams
+}: UseFetchIssueTagsParams) => {
+  const {teams} = useLegacyStore(TeamStore);
+  const {members} = useLegacyStore(MemberListStore);
+
+  const eventsTagsQuery = useFetchOrganizationTags(
+    {
+      orgSlug: org.slug,
+      projectIds,
+      dataset: Dataset.ERRORS,
+      useCache,
+      enabled,
+      keepPreviousData,
+      ...statsPeriodParams,
+    },
+    {}
+  );
+
+  const issuePlatformTagsQuery = useFetchOrganizationTags(
+    {
+      orgSlug: org.slug,
+      projectIds,
+      dataset: 'search_issues',
+      useCache,
+      enabled,
+      keepPreviousData,
+      ...statsPeriodParams,
+    },
+    {}
+  );
+
+  const userTeams = teams.filter(team => team.isMember).map(team => `#${team.slug}`);
+  const usernames: string[] = members.map(getUsername);
+  const nonMemberTeams = teams
+    .filter(team => !team.isMember)
+    .map(team => `#${team.slug}`);
+
+  const suggestedAssignees: string[] = [
+    'me',
+    'my_teams',
+    'none',
+    // New search builder only works with single value suggestions
+    ...(org.features.includes('issue-stream-search-query-builder')
+      ? []
+      : ['[me, my_teams, none]']),
+    ...userTeams,
+  ];
+  const assignedValues: SearchGroup[] | string[] = [
+    {
+      title: t('Suggested Values'),
+      type: 'header',
+      icon: <IconStar size="xs" />,
+      children: suggestedAssignees.map(convertToSearchItem),
+    },
+    {
+      title: t('All Values'),
+      type: 'header',
+      icon: <IconUser size="xs" />,
+      children: [
+        ...usernames.map(convertToSearchItem),
+        ...nonMemberTeams.map(convertToSearchItem),
+      ],
+    },
+  ];
+  const eventsTags = [
+    ...(eventsTagsQuery.data || []),
+    ...(issuePlatformTagsQuery.data || []),
+  ];
+  const eventsTagCollection = eventsTags.reduce<TagCollection>((acc, tag) => {
+    acc[tag.key] = {...tag, kind: FieldKind.TAG};
+    return acc;
+  }, {});
+
+  const additionalTags = builtInIssuesFields(org, eventsTagCollection, assignedValues, [
+    'me',
+    ...usernames,
+  ]);
+
+  const allTags = {
+    ...eventsTagCollection,
+    ...additionalTags,
+  };
+
+  return {
+    tags: allTags,
+    isLoading: eventsTagsQuery.isLoading || issuePlatformTagsQuery.isLoading,
+    isError: eventsTagsQuery.isError || issuePlatformTagsQuery.isError,
+  };
+};
+
+function builtInIssuesFields(
+  org: Organization,
+  currentTags: TagCollection,
+  assigneeFieldValues: SearchGroup[] | string[] = [],
+  bookmarksValues: string[] = []
+): TagCollection {
+  const semverFields: TagCollection = Object.values(SEMVER_TAGS).reduce<TagCollection>(
```

Comment:
The `"RELEASE_STAGE"` field is part of both the builtInTags (line 159 in this file) and in semverFields, but the `getBuiltInTags` function categorizes it as an `"ISSUE_FIELD"` while this function categorizes it as an `"EVENT_FIELD"`. In the return statement, I have deconstructed semver fields after builtInTags, thus resolving the conflict by favoring semver's categorization as an `"EVENT_FIELD"`, but we should maybe resolve this in the `SEMVER_TAGS` or `ISSUE_PROPERTY_FIELDS` declarations